Reorganize examples

jvdydev opened the pull request:
Addresses #383

First draft of what I think makes sense.
Essentially allows to run examples using `--init-directory` (for Emacs 29+, the command is a bit unwieldy for Emacs 28 and earlier) from the `examples/` directory.
I tried to streamline the layout of the examples too.

Feedback appreciated, happy to change stuff around if someone got a better idea :smile:

jeffbowman reviewed (approved):
This looks really good, sorry for the late review.

jeffbowman commented:
> Feedback appreciated, happy to change stuff around if someone got a better idea

Not necessarily better ideas, but you might look at https://github.com/alphapapa/with-emacs.sh for inspiration.  That is a more general approach than what we may need, but there are some interesting ideas there.

jvdydev commented:
Thank you, looks interesting.
I'd probably argue to use `--init-directory` for Emacs >29, but we might want to recommend `with-emacs.sh` for non-Windows pre-Emacs-29 users.
I'll also modify the pre-29 path and test on Windows when I get around to it.

(I'm also marking this WIP for now)

jeffbowman commented:
> I'd probably argue to use --init-directory for Emacs >29,

Absolutely agree!

sthesing reviewed (approved):
Sorry for the late response. I've been a bit out of touch.
This looks really good to me. Thanks for the work you put in!

jvdydev commented:
Fixed some paths that were absolute instead of relative by accident. Also added a ref to alphapapa's with-emacs Script and slightly adapted the example code (mostly to look a bit better in the info buffer). Should be ready for re-review.

jvdydev commented on examples/evil/init.el:
```diff
@@ -0,0 +1,64 @@
+  (customize-set-variable 'evil-escape-delay 0.2)
+
+  ;; Prevent "jj" from escaping any mode other than insert-mode.
+  (defun my/not-insert-state-p ()
```
I think this is a leftover of the examples being rather loose on any form of convention regarding those names (e.g. we have `crafted-startup-example/` prefix in the "base" example).
Since I only really moved around and adjusted the examples to be working, I must've overseen this.

I think a `ce-<example-name>-example-` prefix or similar would be quite nice to have :smile:

jeffbowman commented:
Looks good, just resolve the conflicts and we can merge!

jvdydev commented:
Conflicts should be fixed now :smile:

jeffbowman commented:
Merged!
